linta-android/src/main/java/com/swvl/lint/checks/DuplicateResourceFilesDetector.kt

```diff
@@ -49,42 +48,45 @@ class DuplicateResourceFilesDetector : ResourceXmlDetector() {
     }
 
     override fun visitDocument(context: XmlContext, document: Document) {
+        removeToolsNamespaceAttributes(document.firstChild ?: return)
+
         val stringWriter = StringWriter()
 
         // The transformer will auto-order the elements attributes.
         TransformerFactory.newInstance().newTransformer()
             .transform(DOMSource(document), StreamResult(stringWriter))
 
-        currentDocument = stringWriter.toString()
-
-        removeToolsNamespaceAttributes(document.firstChild ?: return)
+        // Remove extra whitespaces between the elements.
+        val currentDocument = stringWriter.buffer.replace(">[\\s\r\n]*<".toRegex(), "><")
```

Comment:
I think the first regex is the exact same as this one so why did we change it

Reply:
As agreed in chat, we decided to revert this approach, since both approaches will still have caveats such as having something like "4dp " that won't be turned into "4dp", whereas removing all whitespaces shouldn't have as many caveats as this approach.